src/sfdc48/salesforce_sobject_endpoint.bal

```diff
@@ -158,4 +158,44 @@ public type SObjectClient client object {
             return result;
         }
     }
+
+    # Get an object record by Id.
+    #
+    # + sobject - sobject name 
+    # + id - sobject id 
+    # + fields - fields to retrieve 
+    # + return - `json` result if successful else ConnectorError occured
+    public remote function getRecordById(string sobject, string id, string... fields) returns @tainted json|ConnectorError{
+        string path = prepareUrl([API_BASE_PATH, SOBJECTS, sobject, id]);
+        if(fields.length() > 0) {
+            path = path.concat(self.appendQueryParams(fields));            
+        }
+        json response = check self->getRecord(path);
+        return response;
+    }
+
+    # Get an object record by external Id.
+    #
+    # + sobject - sobject name 
+    # + extIdField - external Id field name 
+    # + extId - external Id value 
+    # + fields - fields to retrieve 
+    # + return - `json` result if successful else ConnectorError occured
+    public remote function getRecordByExtId(string sobject, string extIdField, string extId, string... fields) returns @tainted json|ConnectorError{
+        string path = prepareUrl([API_BASE_PATH, SOBJECTS, sobject, extIdField, extId]);
+        if(fields.length() > 0) {
+            path = path.concat(self.appendQueryParams(fields));            
+        }
+        json response = check self->getRecord(path);
+        return response;
+    }
+
+    private function appendQueryParams(string[] fields) returns string{
+        string appended  = "?fields=";
+        foreach string item in fields {
+            appended = appended.concat(item, ",");
```

Comment:
In case if the provided field values is not trimmed. Please check whether it is okay or not.
```suggestion
            appended = appended.concat(item.trim(), ",");
```